app/models/user.rb

```diff
@@ -0,0 +1,51 @@
+class User < ApplicationRecord
+  enum role: {
+         "admin" => 1,
+         "user" => 0,
+       }
+
+  has_secure_password
+  before_create :confirmation_token
+  before_save { self.email = email.downcase }
+  validates :name, presence: true
+  validates :email, presence: true, format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
+  validates :password, presence: true, length: { minimum: 6 }, if: :password_set?
+
+  def verified?
+    if self.verified_at.nil?
+      return false
+    end
+    true
+  end
+
+  def email_activate
+    self.verified_at = Time.now
```

Comment:
can't we simply use
update(email_confirm_token: nil, verified_at: Time.current)